pkg/controller/sparkapplication/submission_test.go

```diff
@@ -398,22 +398,40 @@ func TestPopulateLabels_Driver_Executor(t *testing.T) {
 		t.Fatal(err)
 	}
 	assert.Equal(t, 5, len(driverOptions))
-	assert.Equal(t, fmt.Sprintf(SparkDriverLabelAnnotationTemplate, "app-name", "spark-test"), driverOptions[0])
-	assert.Equal(t, fmt.Sprintf(SparkDriverLabelAnnotationTemplate, "launched-by-spark-operator", strconv.FormatBool(true)), driverOptions[1])
-	assert.Equal(t, fmt.Sprintf(SparkDriverLabelAnnotationTemplate, "submission-id", submissionID), driverOptions[2])
-	assert.Equal(t, fmt.Sprintf(SparkDriverLabelTemplate, AppLabelKey, AppLabelValue), driverOptions[3])
-	assert.Equal(t, fmt.Sprintf(SparkDriverLabelTemplate, DriverLabelKey, DriverLabelValue), driverOptions[4])
+	ExpectedDriverLabels := []string{
+		fmt.Sprintf(SparkDriverLabelAnnotationTemplate, "app-name", "spark-test"),
+		fmt.Sprintf(SparkDriverLabelAnnotationTemplate, "launched-by-spark-operator", strconv.FormatBool(true)),
+		fmt.Sprintf(SparkDriverLabelAnnotationTemplate, "submission-id", submissionID),
+		fmt.Sprintf(SparkDriverLabelTemplate, AppLabelKey, AppLabelValue),
+		fmt.Sprintf(SparkDriverLabelTemplate, DriverLabelKey, DriverLabelValue),
+	}
+
+	for _, label := range driverOptions {
```

Comment:
You don't need to do this. The built-in function `reflect.DeepEqual` can be used to check map equality.

Reply:
`driverOptions` is a slice: `driverOptions, err := addDriverConfOptions(app, submissionID)`. `reflect.DeepEqual` is indeed a good option however, I am not sure it fits what we need. Elements within `driverOptions` may not be inserted in a sorted way since within the function `addDriverConfOptions` we have a non sorted map `driverLabels`. 

Though, back to our test we may have `driverOptions` and `expectedDriverLabels` that have the same labels, however these labels might be sorted in a different way which makes IMHO `reflect.DeepEqual` not appropriate. Please, feel free to propose another alternative.

Reply:
You could sort both slices before calling `DeepEqual`.

Reply:
right :) I changed it.